private/tools/java/com/github/bazelbuild/rules_jvm_external/maven/MavenPublisher.java

```diff
@@ -305,6 +378,83 @@ private static String toHexS(String fmt, String algorithm, byte[] toHash) {
     }
   }
 
+  /**
+   * Attempts to download the file at the given targetUrl. Valid protocols are: http(s), file, and
+   * s3 at the moment.
+   */
+  private static CompletableFuture<Optional<String>> download(String targetUrl) {
```

Comment:
The schemes here aren't the same as `SUPPORTED_SCHEMES`. Is that deliberate?

Reply:
It's a subset. Open to having someone else add the rest as needed.